$MirOS: ports/misc/magicpoint/patches/patch-configure_in,v 1.1.7.1 2005/03/18 15:50:30 tg Exp $
$OpenBSD: patch-configure_in,v 1.1 2003/04/19 13:30:47 espie Exp $
--- configure.in.orig	Wed May 28 05:34:01 2003
+++ configure.in	Sat May 12 22:52:47 2007
@@ -110,6 +110,8 @@ if test "$mgp_use_freetype" = "yes"; the
  			ac_cv_ft_include="$dir/include"
  		elif test -d $dir/include/freetype -a -f $dir/include/freetype/freetype.h; then
  			ac_cv_ft_include="$dir/include/freetype"
+ 		elif test -d $dir/include/freetype1/freetype -a -f $dir/include/freetype1/freetype/freetype.h; then
+ 			ac_cv_ft_include="$dir/include/freetype1/freetype"
 		fi
  		if test "$ac_cv_ft_lib" != "no" -a "$ac_cv_ft_include" != "no"; then
  			LIBS="-L$ac_cv_ft_lib $LIBS"
@@ -160,7 +162,7 @@ if test "$mgp_use_xft2" = "yes"; then
   else
 	AC_MSG_CHECKING(for xft2 and ft2 and fc header..)
 	for dir in /usr/include /usr/include/Xft2 /usr/X11R6/include \
-					/usr/local/include /usr/include/freetype2; do
+			/usr/X11R6/include/freetype2 /usr/local/include /usr/include/freetype2; do
 		if test -f $dir/X11/Xft/Xft.h; then
 			ac_cv_xft2_hdrdir="$dir"
 		fi
@@ -311,10 +313,12 @@ if type pkg-config > /dev/null 2>&1; the
      LIBS="$LIBS `pkg-config --libs libpng`"
      OPTFLAGS="$OPTFLAGS `pkg-config --cflags libpng`"
      AC_DEFINE(USE_PNG)
+     ac_has_found_png=1
   fi
-else
+fi
+if test x"$ac_has_found_png" != x"1"; then
   for i in /usr/pkg /usr/local /usr; do
-	if test -f $i/include/png.h; then 
+	if test -f $i/include/png.h -o -f $i/include/libpng/png.h; then
 		if test -f $i/lib/libpng.a ; then
 			DEPLIBS="$DEPLIBS $i/lib/libpng.a"
 		elif test -f $i/lib/libpng.so ; then
@@ -323,9 +327,12 @@ else
 			break;
 		fi
 		LIBS="$LIBS -L$i/lib"
-		OPTFLAGS="$OPTFLAGS -I$i/include"
+		if test -f $i/include/png.h; then
+			OPTFLAGS="$OPTFLAGS -I$i/include"
+		  else	OPTFLAGS="$OPTFLAGS -I$i/include/libpng"
+		fi
 		AC_CHECK_LIB(png, png_read_image, 
-			[LIBS="$LIBS -lpng"
+			[LIBS="$LIBS -lpng -lz"
 			 AC_DEFINE(USE_PNG)], 
 		[AC_CHECK_LIB(png, png_set_gray_1_2_4_to_8,
 			[LIBS="$LIBS -lpng -lz"
@@ -348,12 +355,12 @@ for i in /usr/pkg /usr/local /usr; do
 		LIBS="$LIBS -L$i/lib"
 		OPTFLAGS="$OPTFLAGS -I$i/include"
 		AC_CHECK_LIB(mng, mng_initialize, 
-			[LIBS="$LIBS -Wl,-rpath,$i/lib -lmng"
+			[LIBS="$LIBS -lmng -llcms -ljpeg -lz"
 			 AC_DEFINE(MNG)], 
 		[AC_CHECK_LIB(mng, mng_readdisplay,
-			[LIBS="$LIBS -lmng -ljpeg"
+			[LIBS="$LIBS -lmng -llcms -ljpeg -lz"
 			 AC_DEFINE(MNG)],
-			[], [-ljpeg])])
+			[], [-llcms -ljpeg -lz])])
 		break
 	fi
 done
@@ -386,20 +393,6 @@ if test $mgp_gif = "yes"; then
 		[], [-L$x_libraries -lX11])], [-L$x_libraries -lX11])])])
 	if test "x$gif_ok" != "xyes"; then
 		echo 'FATAL: You need libungif to use gif images.  pass proper LIBS to configure.'
-		exit 1
-	fi
-	for i in /usr/pkg /usr/local /usr; do
-		if test -f $i/include/gif_lib.h; then
-			gif_h_ok=yes
-			OPTFLAGS="$OPTFLAGS -I$i/include"
-			break
-		fi
-	done
-	if test -f /usr/include/gif_lib.h; then
-		gif_h_ok=yes
-	fi
-	if test "x$gif_h_ok" != "xyes"; then
-		echo 'FATAL: can not find gif_lib.h.  pass proper CPPFLAGS to configure.'
 		exit 1
 	fi
 fi
